cli.js

```diff
@@ -1,84 +1,132 @@
-const commander = require('commander');
-// grab the version to keep things consistent
-const { version } = require('./package.json');
+const yargs = require('yargs');
+const moniker = require('moniker');
+
+const logger = require('./lib/logger');
+const { makeNameValid } = require('./lib/nameUtil');
 const { deployHandler, createHandler, invokeHandler,
-  logHandler, loginHandler, removeHandler, unknownHandler } = require('./lib');
+  logsHandler, loginHandler, removeHandler } = require('./lib');
 
-const actionWrapper = function actionWrapper(action, actionOptions) {
-  return async (options) => {
-    await action({
-      options: Object.assign({}, actionOptions, options, { path: commander.path }),
-      args: commander.args,
-      rawArgs: commander.rawArgs,
-      name: commander.args[commander.args.length - 1]._name,
-    });
-  };
-};
+const handleCommand = async function handleCommand(options, specificHandler) {
+  const numArgs = options._.length;
+  if (numArgs > 1) {
+    yargs.showHelp();
+    logger.error('Too many positional args given for command');
+    process.exit(1);
+  }
 
-// again not the cleanest solution but the practical one while
-// still using commander instead of yargs
-const namedActionHandler = function namedActionHandler(action) {
-  return async (funcName, options) => {
-    await action({
-      options: Object.assign({}, { function: funcName }, options, { path: commander.path }),
-      args: commander.args,
-      rawArgs: commander.rawArgs,
-      name: commander.args[commander.args.length - 1]._name,
-    });
-  };
+  if (!options.function) {
+    yargs.showHelp();
+    logger.error('A valid function name is a required argument');
+    if (options._[0] === 'create') {
+      logger.error('The "create" command allows for the --random(r) flag in lieu of a name');
```

Comment:
I honestly think this is a malfeature that we should get rid of. I just don't see it having any value, and it's one more thing to support.

Reply:
Sure, although as the person using the CLI the most I have to say I use the feature a lot.